README.rst
=============================================
Django Sites Microsoft Authentication Backend
=============================================

.. image:: https://travis-ci.org/gskudder/django_sites_microsoft_auth.svg?branch=master
    :target: https://travis-ci.org/gskudder/django_sites_microsoft_auth
    :alt: Travis CI

.. image:: https://pyup.io/repos/github/gskudder/django_sites_microsoft_auth/shield.svg
    :target: https://pyup.io/repos/github/gskudder/django_sites_microsoft_auth/
    :alt: Updates

.. image:: https://coveralls.io/repos/github/gskudder/django_sites_microsoft_auth/badge.svg?branch=pyup-initial-update
    :target: https://coveralls.io/github/gskudder/django_sites_microsoft_auth?branch=pyup-initial-update
    :alt: Coverage

.. image:: https://api.codeclimate.com/v1/badges/3370bb5f4ecee3af4ee0/maintainability
   :target: https://codeclimate.com/github/gskudder/django_sites_microsoft_auth/maintainability
   :alt: Maintainability

.. image:: https://api.codeclimate.com/v1/badges/3370bb5f4ecee3af4ee0/test_coverage
   :target: https://codeclimate.com/github/gskudder/django_sites_microsoft_auth/test_coverage
   :alt: Test Coverage

Simple app to enable Microsoft Account, Office 365 and Xbox Live authentcation
as a Django authentication backend that is compatible and configurable across multiple sites.


* Free software: MIT license
* Documentation: https://django-microsoft-auth.readthedocs.io.

Features
--------

* Provides Django authentication backend to do Microsoft authentication
  (including Microsoft accounts, Office 365 accounts and Azure AD accounts)
  and Xbox Live authentication.

* Provides Microsoft OAuth client to interfacing with Microsoft accounts

Python/Django support
---------------------

`django_sites_microsoft_auth` follows the same `support cycle as Django <https://www.djangoproject.com/download/#supported-versions>`_,
with one exception: no Python 2 support. If you absoutely need Python 2.7
support, everything should largely already work, but you may need to patch
`microsoft_auth.admin` and/or other files to get it to work.

Supported python versions:  3.5+

Supported Django version: 1.11 LTS, 2.1+

https://docs.djangoproject.com/en/stable/faq/install/#what-python-version-can-i-use-with-django

Note: Even though Django 1.11 LTS supports Python 3.4, I do not and you should
not either. Official support for 3.4 was dropped in March 2019.

Credits
-------

This package was created with Cookiecutter_ and the
`audreyr/cookiecutter-pypackage`_ project template.

.. _Cookiecutter: https://github.com/audreyr/cookiecutter
.. _`audreyr/cookiecutter-pypackage`: https://github.com/audreyr/cookiecutter-pypackage


HISTORY.rst
=======
History
=======

1.0.0 (2020-03-13)
------------------

* Fixes bug with Site note existing
* Add feature to sync user attributes from Azure AD.

0.2.11 (2020-01-13)
-------------------

* Fixes bug where the max_length on a CharField wasn't long enough for all choices to be stored.

0.2.10 (2019-09-24)
-------------------

* Fixes bug in 0.2.9 where cache key was failing.

0.2.9 (2019-09-24)
------------------

* Update cache to work on a per-site basis.

0.2.8 (2019-09-23)
------------------

* Fix bug where migrations were relying on undefined ordering.

0.2.7 (2019-09-23)
------------------

* Fix tests to work with new default for login_enabled.

0.2.6 (2019-09-23)
------------------

* Make sure migrations for sites_microsoft_auth run after migrations for sites framework.

0.2.5 (2019-09-23)
------------------

* Add SiteConfiguration object for all sites during initial migrate.
* Change default for login_enabled to False

0.2.4 (2019-09-23)
------------------

* Fix bug where 'MicrosoftAuthenticationBackend` is not respecting `site` on the MicrosoftAccount model.

0.2.3 (2019-09-23)
------------------

* Fixed bug with `MicrosoftAuthenticationBackend` not restricting users to their respective sites.

0.2.2 (2019-09-23)
------------------

* Updated package name to `sites_microsoft_auth` from `microsoft_auth`

0.2.1 (2019-09-23)
------------------

* Updated documentation.

0.2.0 (2019-09-23)
------------------

* First working release on PyPi


sites_user/migrations/0001_initial.py
# Generated by Django 2.2.5 on 2019-09-23 13:30

import django.contrib.auth.models
from django.db import migrations, models
import django.db.models.deletion
import django.utils.timezone
import sites_microsoft_auth.models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('auth', '__latest__'),
        ('sites', '0002_alter_domain_unique'),
    ]

    operations = [
        migrations.CreateModel(
            name='TestUser',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('password', models.CharField(max_length=128, verbose_name='password')),
                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
                ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
                ('first_name', models.CharField(blank=True, max_length=30, verbose_name='first name')),
                ('last_name', models.CharField(blank=True, max_length=150, verbose_name='last name')),
                ('email', models.EmailField(blank=True, max_length=254, verbose_name='email address')),
                ('is_staff', models.BooleanField(default=False, help_text='Designates whether the user can log into this admin site.', verbose_name='staff status')),
                ('is_active', models.BooleanField(default=True, help_text='Designates whether this user should be treated as active. Unselect this instead of deleting accounts.', verbose_name='active')),
                ('date_joined', models.DateTimeField(default=django.utils.timezone.now, verbose_name='date joined')),
                ('username', models.CharField(help_text='Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.', max_length=150, unique=True, validators=[sites_microsoft_auth.models.UnicodeSpaceUsernameValidator], verbose_name='username')),
                ('groups', models.ManyToManyField(blank=True, help_text='The groups this user belongs to. A user will get all permissions granted to each of their groups.', related_name='user_set', related_query_name='user', to='auth.Group', verbose_name='groups')),
                ('site', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='sites.Site')),
                ('user_permissions', models.ManyToManyField(blank=True, help_text='Specific permissions for this user.', related_name='user_set', related_query_name='user', to='auth.Permission', verbose_name='user permissions')),
            ],
            options={
                'abstract': False,
            },
            managers=[
                ('objects', django.contrib.auth.models.UserManager()),
            ],
        ),
    ]
